### ISSUE TITLE
Translation

### ISSUE THREAD

**Issue by mried** [remig/web_lic#17]
Hi,

I had a look on the `contextMenu.js` to extract all the strings for translation. This is the structure of a language file now:
* Some general texts like `ok`, `cancel` or `step`
* Subsection `navbar`
* Subsection `dialog`
* Subsection `undo`
* Subsection `status_bar`

We could add the context menu strings into a new subsection `context_menu`:
```json
"context_menu": {
    "add_annotation": "Add Annotation",
    "add_annotation_line": "Line",
    "<and so on>"
}
```

Another idea is to refactor the structure a bit: Add a subsection `action` and place all context menu strings there as well as the nav bar strings. We could also place the undo and status bar strings there. This leads to the following:
```json
"action": {
    "add_annotation": {
        "name": "Add Annotation",
        "line": {
            "name": "Line"
        }
    },
    "delete_item": {
        "name": "Delete",
        "undo_@mf": "Delete ${item}",
        "success_message_@mf": "${item} deleted"
    },
    "<and so on>"
}
```
If we decide to add a tooltip or other action related texts, this can be added here very simple.

I tend to the second solution because of its clearer structure (in my eyes). Everything which belongs to a specific action is in one place.

Another thing: I'd like to add a subsection `glossary` which takes all the general texts like `step`, `callout` and so on (`ok`, `cancel` and `reset` should be moved into the `dialog` subsection).

What do you think?

PS: I'd like to use this issue to discuss all upcoming points for the translation. Feel free to close the issue, we can still use it afterwards.

**Comment by remig** [comment]
@mried made a handful of fixes to the translation logic, moved all remaining hardcoded strings into the translation table, and added a full German translation.  I've documented how to [create a new translation](https://github.com/remig/web_lic/wiki/Translating-Lic). I believe the translation feature is now complete - it's looking great!

**Comment by mried** [comment]
Hi,
great to hear! I wonder I didn't get an email about your comment... And sorry for the bugs I've created.

There is still one thing left I couldn't figure out: For some strings it is necessary to reload the page after changing the language to make them show up in the newly selected language. Especially those of the side bar which is shown at the template page.

I'll create a new pull request in a minute containing translations for the new strings.

**Comment by remig** [comment]
I just pushed a fix to correctly update the template page labels when the language changes. Thanks for catching that! If there are still some strings that don't update when you change the language, let me know; it's a bug with how / when that particular string refreshes.

I also found one more untranslated label: if you select the template page's rotate icon, the 'Arrow' title was untranslated.  I added a new translation string for that.
